middleware/dnstap/setup.go

```diff
@@ -26,30 +28,43 @@ func wrapSetup(c *caddy.Controller) error {
 	return nil
 }
 
-func parseConfig(c *caddyfile.Dispenser) (path string, full bool, err error) {
+func parseConfig(c *caddyfile.Dispenser) (path string, socket, full bool, err error) {
```

Comment:
Shouldn't keep adding return parameters in this function. Either start an dnsOpt struct which has all these or, if possible, fill out a *dnstap structure with the correct elements.